scalding-core/src/main/scala/com/twitter/scalding/DescribedExecutionApp.scala

```diff
@@ -0,0 +1,69 @@
+package com.twitter.scalding
+
+sealed abstract class DescribedArg {
+  def key: String
+  def description: String
+}
+
+case class RequiredArg(key: String, description: String) extends DescribedArg
+case class OptionalArg(key: String, description: String) extends DescribedArg
+case class ListArg(key: String, description: String) extends DescribedArg
+case class BooleanArg(key: String, description: String) extends DescribedArg
+
+object ArgHelp {
+  /**
+    * Describe the Arguments of this Execution.  By running --help the args will output
+    * and the execution will end
+    * @param describedArgs List of Argument Descriptions
+    * @param ex Input Execution
+    * @return Output Execution
+    */
+  def describe(describedArgs: Seq[DescribedArg], ex: Execution[Unit]): Execution[Unit] = {
```

Comment:
A second method sounds reasonable, or a flag here. None of the information is passed onto the output execution here so chaining is likely difficult.